apps/history/declarativehistorymodel.cpp

```diff
@@ -66,6 +65,23 @@ void DeclarativeHistoryModel::remove(const QString &url)
     }
 }
 
+void DeclarativeHistoryModel::add(const QString &url, const QString &title)
+{
+    QDateTime time = QDateTime::currentDateTimeUtc();
+    int id = DBManager::instance()->addHistoryEntry(url, title, time);
```

Comment:
I meant this single line should be enough. When you start typing to the search field, model updates on every single character and dbworker emits historyAvailable.

Problem with this approach is that appending to the end can be wrong place as sorting comes from dbworker. It should be inserted to the right place if any. Hence, I'd not add anything extra here rather just call addHistoryEntry(...).

If you think that's worth to update model here, you could store current search term to member variable (see search) and call search(m_searchTerm);